config/crd/patches/validation_in_linodemachines.yaml

```diff
@@ -0,0 +1,17 @@
+# The following patch adds additional constraints after the built-in name validation for the CRD
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/metadata/properties
+  value:
+    name:
+      type: string
+      x-kubernetes-validations:
+      - rule: 3 <= size(self) && size(self) <= 64
+        message: linode instance labels must be between 3..64 characters
+      # TODO: Combine these into one regex to validate ALL constraints on Linode instance labels at once?
+      # https://www.linode.com/docs/api/linode-instances/#linode-create
+      # - rule: self.contains('^[a-z0-9](.*[a-z0-9])*$')
+      #   message: linode instance labels must begin and end with an alphanumeric character
+      # - rule: self.contains('^[-_.[:alnum:]]*$')
+      #   message: linode instance labels may only consist of alphanumeric characters, hyphens (-), underscores (_) or periods (.)
+      - rule: "!(self.contains('--') || self.contains('__') || self.contains('..'))"
+        message: linode instance labels cannot have two hyphens (--), underscores (__) or periods (..) in a row
```

Comment:
label is a bit of an overloaded term here that I'm not sure will be completely obvious to end users how it correlates to the `name` of a resource. could we possibly reword these to mention it is just a requirement on the naming of the resource itself?

Reply:
If it helps, the formatting of this validation within the CRD means that API server will automatically reference the `metadata.name` field in the error message like so:
```sh
Error from server (Invalid): error when creating "STDIN": LinodeMachineTemplate.infrastructure.cluster.x-k8s.io "bad--name-control-plane" is invalid: metadata.name: Invalid value: "string": linode instance labels cannot have two hyphens (--), underscores (__) or periods (..) in a row
Error from server (Invalid): error when creating "STDIN": LinodeMachineTemplate.infrastructure.cluster.x-k8s.io "bad--name-md-0" is invalid: metadata.name: Invalid value: "string": linode instance labels cannot have two hyphens (--), underscores (__) or periods (..) in a row
``` 

I can still change this though if you feel it's unclear 🙂

Reply:
that is fairly clear, I am still on the fence on using the label terminology but it is more of a nit at this point so I'm good with that not being a blocking concern